Add data format and modalities sections (#413)
AlexsLemonade/scpca-docs: docs/download_options.md

# Download options

You can obtain Portal data either by downloading a single project, creating a custom dataset with a selection of projects and/or samples, or by choosing one of the Portal-wide download options.
For full information about the files you can download from the Portal, see {ref}`the Downloadable Files page<downloadable_files:Downloadable files>`.

This page describes the different options available to you when downloading Portal data.

## Data format

We provide all single-cell/nuclei expression data in both [`SingleCellExperiment` objects (`.rds` files)](#singlecellexperiment-downloads) for use in R, and as [`AnnData` objects (`.h5ad` files)](#anndata-downloads) for use in Python.
You can learn more about using these object types from our FAQ sections on {ref}`using the provided RDS files<faq:How do I use the provided RDS files in R?>` and {ref}`using the provided H5AD files<faq:How do I use the provided H5AD files in Python?>`.

> Note that only one data format is currently supported for a single download, including when {ref}`downloading custom datasets<downloadable_files:Custom datasets>`.

Author: I thought about this, but I couldn't find a way to transition into it that I liked. The best I am now coming up with is something like this (in bold for emphasis here but wouldn't be in docs), but I don't love this either. Do you have an idea?

We provide all single-cell and single-nuclei expression data in both `SingleCellExperiment` objects (`.rds` files) for use in R **(default)**, and as `AnnData` objects (`.h5ad` files) for use in Python.

To obtain data in both `SingleCellExperiment` and `AnnData` formats, you will need to download these file formats separately.

In addition, note that `AnnData` files are not available for projects with multiplexed libraries all data on the Portal, {ref}`as described here:<download_files:Multiplexed sample libraries>`.

## Modalities

Besides single-cell/nuclei expression, many samples in the Portal have additional sequencing modalities including CITE-seq, bulk RNA-seq, and spatial transcriptomics.

When downloading a project with spatial transcriptomics data, you will have the option to select either the "Single-cell" or "Spatial" modality.
Selecting "Single-cell" will provide you with the single-cell/nuclei expression data only, and selecting "Spatial" will provide you with the spatial data only.
If available, CITE-seq data will be automatically included with the "Single-cell" modality only.
Only one modality can be selected at a time.

> If the project also has bulk RNA-seq data, you can choose to include it in your download for either modality selection.

Author: Or, maybe just state that bulk is automatically included with a project download with either modality if there is bulk, but we don't talk about the Custom Dataset behavior here.

> If the project also has bulk RNA-seq data, you can choose to include it in your download for either modality selection.

Reviewer: > Or, maybe just state that bulk is automatically included with a project download with either modality if there is bulk, but we don't talk about the Custom Dataset behavior here.

I think bulk is only included with single-cell downloads and not spatial. Since this page is focused on download options, and choosing to download bulk when creating a custom dataset is a download option we do want to mention it. Maybe we should have a separate section describing the download options for bulk?

> If the project also has bulk RNA-seq data, you can choose to include it in your download for either modality selection.

Reviewer: Actually, maybe we just rephrase what you have to be: 

```suggestion
When selecting samples that have associated bulk RNA-seq, you will have the option to include the bulk RNA-seq in your download. 
If downloading an entire project using the `Download Now` button, any associated bulk RNA-seq data for samples in that project will automatically be included. 
```

> If the project also has bulk RNA-seq data, you can choose to include it in your download for either modality selection.

Author: I'm good with this

# Download options

You can obtain Portal data either by downloading a single project, creating a custom dataset with a selection of projects and/or samples, or by choosing one of the Portal-wide download options.
For full information about the files you can download from the Portal, see {ref}`the Downloadable Files page<download_files:Downloadable files>`.

This page describes the different options available to you when downloading Portal data.

## Data format

> We provide all single-cell and single-nuclei expression data in both `SingleCellExperiment` objects (`.rds` files) for use in R, or as `AnnData` objects (`.h5ad` files) for use in Python.

Reviewer: ```suggestion
We provide all single-cell and single-nuclei expression data in both `SingleCellExperiment` objects (`.rds` files) for use in R, or as `AnnData` objects (`.h5ad` files) for use in Python.
The default format for all samples on the Portal with single-cell and single-nuclei expression is set to `SingleCellExperiment (R)`. 
```
I think you could mention the default up front.

For full information about the files you can download from the Portal, see {ref}`the Downloadable Files page<download_files:Downloadable files>`.

This page describes the different options available to you when downloading Portal data.

## Data format

We provide all single-cell and single-nuclei expression data in both `SingleCellExperiment` objects (`.rds` files) for use in R, or as `AnnData` objects (`.h5ad` files) for use in Python.
The default format for all samples on the Portal with single-cell and single-nuclei expression is set to `SingleCellExperiment (R)`.
You can learn more about using these object types from our FAQ sections on {ref}`using the provided RDS files<faq:How do I use the provided RDS files in R?>` and {ref}`using the provided H5AD files<faq:How do I use the provided H5AD files in Python?>`.

Only one data format is currently supported for a single download, including when {ref}`downloading custom datasets<download_files:Custom datasets>`.
To obtain data in both `SingleCellExperiment` and `AnnData` formats, you will need to download these file formats separately.

> In addition, note that expression data for multiplexed libraries is only available in `SingleCellExperiment`  format, {ref}`as described here<download_files:Multiplexed sample libraries>`.

Reviewer: ```suggestion
In addition, note that expression data for multiplexed libraries is only available in `SingleCellExperiment` format, {ref}`as described here<download_files:Multiplexed sample libraries>`.
```

## Modalities

Besides single-cell/nuclei expression, many samples in the Portal have additional sequencing modalities including CITE-seq, spatial transcriptomics, and bulk RNA-seq.

In particular, there are two modality options that you may see when {ref}`creating a custom dataset to download<download_files:Custom datasets>` or when {ref}`downloading a full project with the "Download Now" button<download_files:Project downloads>`: "Single-cell" and "Spatial."

By default, the "Single-cell" modality will be selected for all single-cell and single-nuclei RNA-seq samples and/or projects.
Selecting this download option will provide you with the gene expression data from single-cell or single-nuclei samples and/or projects.
If available, CITE-seq expression data will also be included.

> If a sample or project has spatial transcriptomic data, you will also have the option to select the "Spatial" modality for download.
> Selecting "Spatial" will provide you with the spatial transcriptomic data only.
> This option is also available when downloading any projects that have samples with spatial transcriptomic data.

Reviewer: ```suggestion
If a sample has spatial transcriptomic data, you will also have the option to select the "Spatial" modality for download.
Selecting "Spatial" will provide you with the spatial transcriptomic data only.
This option is also available when downloading any projects that have samples with spatial transcriptomic data.
```

Author: The reason I have "project" here is because of this (from the top of `download-options-cart-summary` in figma): 
![Screenshot 2025-05-09 at 3 46 33 PM](https://github.com/user-attachments/assets/bb80a84a-6f7c-4785-943a-2db391870572)

Because you can select modality for both samples and projects, I have a slight preference to keep this in, but feel free to talk me out of it!!

Reviewer: I was editing it because you have a sentence that talks specifically about projects. Alternatively:

```suggestion
If a sample or project has spatial transcriptomic data, you will also have the option to select the "Spatial" modality for download.
Selecting "Spatial" will provide you with the spatial transcriptomic data only.
```

Author: Oh lord, happy Friday.
